server/src/main/java/org/elasticsearch/ingest/IngestStats.java

```diff
@@ -76,6 +81,9 @@ public static IngestStats read(StreamInput in) throws IOException {
                 var processorName = in.readString();
                 var processorType = in.readString();
                 var processorStat = readStats(in);
+                // pass these name and type through the local names and types cache to canonical-ize them
+                processorName = namesAndTypesCache.computeIfAbsent(processorName, k -> k);
```

Comment:
NIT: Lets use `Function.identity()` :P no need to blow up the metaspace more than necessary :)

Reply:
Sure thing, 4d9cadc0b71defbe093b771d53f6066aabaa67fb.